### ISSUE TITLE
TypeError: Cannot read property 'statusText' of undefined

### ISSUE THREAD

**Issue by pwdelbloomboard** [kbravh/tweet-to-markdown#7]
Type Error 

I attempted to run this command on a long thread:

```
$ ttm --bearer "TTM>XXXXX" https://twitter.com/iotmpls/status/1555601890116788225 -a -t
```

* So as you can see, I'm saving the images locally and attempting to run through and write up as long of a markdown file that I could. However, partway through (118 tweets through), it failed with:

`$ ttm --bearer "TTM>XXXXX" https://twitter.com/iotmpls/status/1555601890116788225 -a -t
/Users/patrick/.config/yarn/global/node_modules/tweet-to-markdown/dist/main.js:14961
        panic(source.red(error.response.statusText));
                                        ^

TypeError: Cannot read property 'statusText' of undefined
    at /Users/patrick/.config/yarn/global/node_modules/tweet-to-markdown/dist/main.js:14961:41
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)`

* Moreover, while I did get a large number of images downloaded, no markdown file was authored.

A couple points:

1. I believe the expected functionality from a user standpoint would be that even if there is an error, the markdown file should have been at least created. Now what I have is a folder full of images, with no markdown file. So perhaps the user ran out of credits, that would be an acceptable outcome, because it's up to the user to have enough credits to finish the tweet string. However then the error handling should say, "not enough credits!  Finishing thread."

I don't know much about how node works, but the equivalent behavior I would expect in bash psuedocode would be something like:

```
# start a file
touch tweetstringfile.md

# for each tweet starting at tweet 0 through n
for TWEET in seq 0 n
do
    # if ran out of credits
    echo "you ran out of credits! Closing and saving markdown file thread."
    # pull each tweet
    echo $TWEET >> tweetstringfile.md
done
```

Where essentially the >> is "appending" to the bottom of the file.

Better yet, prompt the user ahead of time to understand that the number of the tweets vs. remaining credits is acceptable. Warning! This will use up all of your remaining credits and will not finish the .md file!

2. All of the above being said, I believe the error was actually just an API error, which came from here in your code:

https://github.com/kbravh/tweet-to-markdown/blob/main/src/util.ts#L141

So basically, the TTM API didn't respond in that once instance, and without the proper error handling and logic built in to your program, it appears that your program collects everything in, "string"

https://github.com/kbravh/tweet-to-markdown/blob/main/src/util.ts#L153

and then writes it to a markdown file all at once in the end:

https://github.com/kbravh/tweet-to-markdown/blob/main/src/util.ts#L332

So basically if any error occurs along the way, then a markdown file never gets written.

Does that sound correct to you? What are your thoughts?

**Comment by pwdelbloomboard** [comment]
Further clarification on the above, I did not run out of credits, I used 120 out of 200 credits thus far when the error showed up.

**Comment by kbravh** [comment]
Hi @pwdelbloomboard, I think the error may have been caused by rate limiting, but you're right in that the error handling is not ideal in this case. I've reset your usage count back to 0, and I'm going to get to work on cleaning up the error handling, and writing partial files even if something does go wrong an error occurs instead of simply failing.

**Comment by kbravh** [comment]
@pwdelbloomboard I've changed the tool so that partial files will be written if an error occurs while downloading a thread, and I've added much better error handling and tests in general. The newest version (2.1.1) includes these fixes.
